docs/docs/rest-api/public/api/v2/groups.raml

```diff
@@ -0,0 +1,376 @@
+###
+# Get the root group
+#
+get:
+  description:
+    Get the root group with all applications and all transitive child groups.
+  is: [ secured ]
+  responses:
+    200:
+      description: The root group with all transitive dependencies.
+      body:
+        application/json:
+          example: !include examples/group.json
+          schema: Group
+
+###
+# Update the root group
+#
+put:
+  description:
+    Change parameters of a deployed application group.
+    The new group parameters get applied.
+
+
+    * Changes to application parameters will result in a restart of this application.
+
+    * A new application added to the group is started.
+
+    * An existing application removed from the group gets stopped.
+
+
+    If there are no changes to the application definition, no restart is triggered.
+    During restart marathon keeps track, that the configured amount of minimal running instances are _always_ available.
+
+    This method alows 2 special modes for the update operation>
```

Comment:
Add: When one of `version` or `scaleBy` are set, nothing else than a version change or transitive instance count scaling will be applied. If both `version` and `scaleBy` are set, only a version rollback will be performed – the scaleBy value will not be applied.

- There should be tests for this implementation detail
- It might be better to throw validation errors than to silently drop body parts by performing `versionChange orElse scaleChange getOrElse update.apply`.